Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -2456,6 +2449,24 @@ checkDuplicateModules pkg =
 -- * Utils
 -- ------------------------------------------------------------
 
+toDependencyVersionsMap :: (PackageDescription -> [Dependency]) -> GenericPackageDescription -> Map PackageName VersionRange
+toDependencyVersionsMap selectDependencies pkg = case typicalPkg pkg of
+      Right (pkgs', _) ->
+        let
+          self :: PackageName
+          self = pkgName $ package pkgs'
+        in
+        Map.fromListWith intersectVersionRanges $ do
+          Dependency pname vr _ <- selectDependencies pkgs'
+          if pname == self then
+            []
+          else [(pname, vr)]
```

Comment:
I think this would be easier to read with a list comprehension:
```haskell
  Map.fromListWIth intersectVersionRanges
    [ (pname, vr) 
    | Dependency pname vr _ <- selectDependencies pkgs'
    , pname /= self
    ]
```

Reply:
I changed thsi to the list comprehension like suggested.